src/WebJobs.Script/ScriptHostBuilderExtensions.cs

```diff
@@ -404,6 +404,23 @@ internal static void ConfigureApplicationInsights(HostBuilderContext context, IL
                         o.EnableDependencyTracking = false;
                     });
                 }
+
+                if (SystemEnvironment.Instance.IsApplicationInsightsAgentEnabled())
```

Comment:
For the dotnet-isolated devs (and others) -- I've talked with @RohitRanjanMS and we can re-use this logic, but instead of using a Java-specific environment variable, we'd use a capability that the worker can send to the host. The logic to filter will be the same.

I can follow up with that work. I think that'll be more general (and not require an app setting, etc), but we don't really need to block this PR on that.